Tests/GRPCHTTP2CoreTests/Client/Resolver/DNSResolverTests.swift

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCHTTP2Core
+import Testing
+
+@Suite("DNSResolver")
+struct DNSResolverTests {
+  @Test("Resolve hostname")
+  func resolve() async throws {
+    let expectedResult: [SocketAddress] = [
+      .ipv6(host: "::1", port: 80),
+      .ipv4(host: "127.0.0.1", port: 80),
+    ]
+
+    let result = try await DNSResolver.resolve(host: "localhost", port: 80)
```

Comment:
I don't know if we can rely on the ordering of the results. Probably worth resolve "::1" and "127.0.0.1" directly and checking we get back what we expect.